data-prepper-plugins/mutate-event/src/main/java/com/amazon/dataprepper/plugins/processor/mutateevent/CopyValueProcessor.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.processor.mutateevent;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.processor.AbstractProcessor;
+import com.amazon.dataprepper.model.processor.Processor;
+import com.amazon.dataprepper.model.record.Record;
+
+import java.util.Collection;
+
+@DataPrepperPlugin(name = "copy_value", pluginType = Processor.class, pluginConfigurationType = CopyValueProcessorConfig.class)
+public class CopyValueProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
+    private final CopyValueProcessorConfig config;
+
+    @DataPrepperPluginConstructor
+    public CopyValueProcessor(final PluginMetrics pluginMetrics, final CopyValueProcessorConfig config) {
+        super(pluginMetrics);
+        this.config = config;
+    }
+
+    @Override
+    public Collection<Record<Event>> doExecute(final Collection<Record<Event>> records) {
+        for(final Record<Event> record : records) {
+            final Event recordEvent = record.getData();
+            final String key = config.getFromKey();
+            final String newKey = config.getToKey();
+            if (!key.equals(newKey)
```

Comment:
If the keys are equal we don't need to loop through every event. Let's perform this check prior to the for loop.